### ISSUE TITLE
Default values from fmf are not propagated and used in filters

### ISSUE THREAD

**Issue by Jakuje** [psss/tmt#450]
The `test show` correctly shows the test as enabled:
```
$ mkdir /tmp/enabled && cd /tmp/enabled && tmt init -t base && tmt test show
/tests/example
     summary Concise summary describing what the test does
        test ./test.sh
        path /tests/example
   framework beakerlib
      manual no
     require beakerlib
    duration 5m
     enabled yes
      result respect
```

Filtering based on explicit keys in the test plan is working correctly:
```
$ tmt tests ls --filter "framework:beakerlib"
/tests/example
```

But we are unable to filter based on this default value:
```
$ tmt tests ls --filter "enabled:true"
[empty]
```
(should list the above test)

**Comment by WOnder93** [comment]
One more related issue: The filter "enabled" only works with values "True" or "False" (exact case), but using other aliases that are valid in test metadata (e.g. "yes" or "true") doesn't work. That is confusing and the same values should work in metadata and filters.

**Comment by psss** [comment]
Yeah, these are caused by the very simple `filter()` implementation, see `pydoc fmf.filter` for more info, which is called in the fmf metadata before the tmt specification (such as defaults) are applied. We should move the filtering to a later phase when all values are correctly initialized.

**Comment by FrNecas** [comment]
Seems to be related to #469 , should be resolved by #521 once its merged, although listing and filters are case sensitive so the following would be necessary (note the `True`):
```
$ tmt test ls --filter "enabled:True"
/tests/example
```
Fmf allows case-insensitive filtering, perhaps we should add an option to tmt that would support it (or maybe even make it the default? Otherwise filtering based on boolean values could be slightly confusing to users). What do you think @psss ?
